andela/ah-alpha
# #162948907 generate and return jwt tokens

#### What does this PR do?
Users should receive a JWT on successful Registration 

#### Description of Task(s) to be completed?
return a JWT to a user on successful registration
generate and return a JWT when a user logs in

#### How should this be manually tested?
* Pull the branch `git fetch origin ft-return-token-162948907:ft-return-token-162948907`
* Checkout to the branch `git checkout ft-return-token-162948907`
* Install dependencies `pip3 install -r requirements.txt`
* Run the server `python3 manage.py runserver`
* Use postman to send a `POST` request to `http://127.0.0.1:8000/api/users/`
* The data should be in the format of:
```
{   "user":{
    "username": "dann",
    "email": "dann@dan.com",
    "password": "county001"
    }
 }
```
* Remember to select `application/json` as the data type
* The response should be a json object containing the user's details as well as the jwt token
```
{
    "user": {
        "email": "dann@dan.com",
        "username": "dann",
        "token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJ1c2VyX2lkIjoxNiwidXNlcm5hbWUiOiJkYW5uQGRhbi5jb20iLCJleHAiOjE1NDc1NzgyMDgsImVtYWlsIjoiZGFubkBkYW4uY29tIn0.n8mW-ncEXc37EwALIlQgF6CVyMtdjKWuowy6SYjamPA"
    }
}
```


#### Any background context you want to provide?
* This project uses the latest version of python3. Please download it from [here](https://www.python.org/downloads/)
* You may as well use pip instead of pip3 if you have either only python3 installed on your computer or you have it set as the default version of python
* Postman can be downloaded from [here](https://www.getpostman.com/downloads/)
* This token needs to be sent in the headers of subsequent requests to protected endpoints
* The header prefix expected is `Bearer`. On postman, this would hence be sent as a Bearer Token
* I also downgraded django-rest-swagger from v2.2.0 to v2.1.2 due to a bug where the authorization headers aren't passed in the request. It is well described [here](https://github.com/marcgibbons/django-rest-swagger/issues/762).

#### What are the relevant pivotal tracker stories?
[#162948907](https://www.pivotaltracker.com/story/show/162948907)
